File: vulnerable/php/vuln_example.php
<?php
$DB_USER = "root"; // hardcoded
$DB_PASS = "secret123"; // hardcoded
$mysqli = mysqli_connect("localhost", $DB_USER, $DB_PASS, "test");

$name = $_GET['name'];
$code = $_GET['code'];

// SQL injection via concatenation
$query = "SELECT * FROM users WHERE name = '$name'";
$result = mysqli_query($mysqli, $query);

// Dangerous eval
eval($code);

echo "Hello " . $name; // Potential XSS
?>

File: vulnerable/python/vuln_example.py
import sqlite3
import hashlib

DB_USER = "admin"  # hardcoded credential
DB_PASS = "password123!"  # hardcoded credential


def insecure_login(username, password):
    # Weak crypto (MD5)
    hashed = hashlib.md5(password.encode()).hexdigest()

    # SQL injection via string concatenation
    conn = sqlite3.connect("test.db")
    cur = conn.cursor()
    query = "SELECT * FROM users WHERE name = '" + username + "' AND pass_hash = '" + hashed + "'"
    try:
        cur.execute(query)  # Non-parameterized query
        for row in cur.fetchall():
            print(row)
    except Exception:
        # Empty catch (code smell)
        pass
    finally:
        conn.close()


def bad_practices():
    temp = 42  # unused variable (code smell)
    code = input("Enter code to eval: ")
    eval(code)  # dangerous use of eval (security hotspot)
